src/components/Greeting.js

```diff
@@ -1,64 +1,40 @@
-import React, { useEffect, useState } from 'react';
-import { connect } from 'react-redux';
-import PropTypes from 'prop-types';
-import { fetchRandomMessage } from '../redux/action';
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { fetchRandomGreeting } from '../redux/greetingSlice';
 
-const Greeting = ({ randomMessage, fetchRandomMessage }) => {
-  const [isLoading, setIsLoading] = useState(true);
-  const [countdown, setCountdown] = useState(10);
+function Greeting() {
+  const dispatch = useDispatch();
+  const { greeting, isLoading, error } = useSelector((state) => state.greetings);
 
   useEffect(() => {
-    const fetchMessage = async () => {
-      try {
-        await fetchRandomMessage();
-        setIsLoading(false);
-      } catch (error) {
-        throw new Error('Error fetching data:', error);
-      }
-    };
-
-    fetchMessage();
-  }, [fetchRandomMessage]);
-
-  useEffect(() => {
-    if (countdown === 0) {
-      fetchRandomMessage();
-      setCountdown(10);
-    } else {
-      const timer = setTimeout(() => setCountdown(countdown - 1), 1000);
-      return () => clearTimeout(timer);
-    }
-    return undefined;
-  }, [countdown, fetchRandomMessage]);
+    dispatch(fetchRandomGreeting());
+  },
+  [dispatch]);
 
   if (isLoading) {
-    return <div className="loading">Loading...</div>;
+    return (
+      <>
+        <div className="loading">...isLoading</div>
+      </>
+    );
+  }
+
+  if (error) {
+    return (
+      <>
+        <div className="loading">
+          Something is wrong during fetching the data
+        </div>
+      </>
+    );
   }
 
   return (
-    <div className="greeting-content">
-      <h1>Greeting of the day</h1>
-      <p>
-        &quot;
-        {randomMessage}
-        &quot;
-      </p>
-      <h2>
-        Greeting changes in:
-        {' '}
-        <span>{countdown}</span>
-      </h2>
-    </div>
+    <>
+      <div className="title">Random Greeting</div>
+      <div className="greeting">{greeting}</div>
+    </>
   );
-};
-
-Greeting.propTypes = {
-  randomMessage: PropTypes.string.isRequired,
-  fetchRandomMessage: PropTypes.func.isRequired,
-};
-
-const mapStateToProps = (state) => ({
-  randomMessage: state.randomMessage,
-});
+}
 
-export default connect(mapStateToProps, { fetchRandomMessage })(Greeting);
+export default Greeting;
```

Comment:
- [x] Good Job for addressing the suggested changes requested by the previous reviewer, But Kindly make sure that the React app uses the url and displays a random greeting , Because When I try to test the react app to display the random message by using the backed message Api , the react app display the below messages instead of display the random message I crested on the backend app, So kindly Fix the issues.
- Your React App using the url for backend 
![image](https://github.com/leloufadel/hello-react-frontend/assets/93598189/350060d9-48b4-4b95-91ef-a32733cdd5d3)

- Because when I checked Check the network requests in your browser's developer tools when I load the application. As shown with the below screenshot I Verify that the request to 'http://127.0.0.1:3000/api/messages/random' is being made and displayed the random message.
![image](https://github.com/leloufadel/hello-react-frontend/assets/93598189/ad9d0541-5ae4-4340-b75c-ba138940a1e6)

- So Kindly Fix the issues

Reply:
![image](https://github.com/leloufadel/hello-react-frontend/assets/122641237/af6a7d48-e179-4b39-9926-e3643e63984d)